### PR TITLE
easyengine 4.0.1 (new formula)

### PR DESCRIPTION
**Pull request by mbtamuli**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Signed-off-by: Mriyam Tamuli <mbtamuli@gmail.com>

### CODE REVIEW

**Comment by mbtamuli** [comment]
@lembacon The build seems to have passed. Do you need anything else from my end to merge this?

**Comment by PatelUtkarsh** [comment]
@SMillerDev 
> Does this require both PHP and Docker?

Yep. The project is in PHP language so PHP is a dependency. 
Also Docker is required as the entire project revolves around it.

File: Formula/easyengine.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    EasyEngine requires Docker to work.
+
+    You can install Docker using:
+
+      brew cask install docker
+  EOS
+  end
+
+  test do
+    system bin/"ee cli version"
```
[the remark this answers is not in the document]
Comment (PatelUtkarsh) [reply Formula/easyengine.rb]: Since we have dependency on docker for all current feature set, other feature test case will not able to pass in CI.

We can add more test-cases in future when we have more command which don’t depend on docker. @SMillerDev

[the remark this answers is not in the document]
Comment (javian) [reply Formula/easyengine.rb]: I don't think this would be suitable for the core tap in that case especially since we can't get it to work without doing additional commands and not have it tested. You already have a viable way of installing it as shown on the website so I would stick to that for now imo.

**Comment by SMillerDev** [comment]
@Homebrew/core what is the opinion on formula requiring docker and not being usable otherwise?

**Comment by MikeMcQuaid** [comment]
@SMillerDev I'm not sure I understand why a `.phar` or a macOS executable would require Docker. If you need Docker anyway: why aren't you installing it as part of the Dockerfile?

**Comment by mbtamuli** [comment]
@MikeMcQuaid The EasyEngine project itself is a written in PHP. It doesn't require docker to run. Put it other way, if docker is not installed, it will just print a message informing that and exit.

I guess our comments were not clear enough and caused confusion. Apologize for that.

EasyEngine itself is a simple phar file like wp-cli - https://github.com/Homebrew/homebrew-core/blob/master/Formula/wp-cli.rb

In fact, EasyEngine has its roots in [WP-CLI](https://wp-cli.org/) project.

EasyEngine has some part which runs even without Docker. We will update PR to have those part of brew's test cases.

**To summarize:**
Docker is not hard dependency for EasyEngine itself. If Docker doesn't exist EasyEngine won't crash.

**Comment by MikeMcQuaid** [comment]
Ok. I think the caveats can be removed then, thanks.

**Comment by mbtamuli** [comment]
@MikeMcQuaid Thanks. I have updated the PR. Removed the caveats and added more tests.

CC: @SMillerDev

**Comment by mbtamuli** [comment]
@MikeMcQuaid @SMillerDev Not sure what went wrong. The build failed for the [mojave](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/33937/version=mojave/) configuration due to what seems like network issue.

**Comment by javian** [comment]
Yes there's a currently an issue with the CI which we haven't resolved yet.

**Comment by javian** [comment]
@BrewTestBot test this please

**Comment by mbtamuli** [comment]
@javian Thanks running the tests again. They seems to have passed. :slightly_smiling_face:

File: Formula/easyengine.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    bin.install "easyengine.phar" => "ee"
+  end
+
+  test do
+    system bin/"ee config set locale hi_IN"
+    output = shell_output("#{bin}/ee config get locale")
+    assert_match "hi_IN", output
+
+    output = shell_output("#{bin}/ee cli version")
+    assert_match "EE 4.0.1", output
```
Comment (javian) [inline Formula/easyengine.rb]: You should be able to replace 4.0.1 with #{version}

**Comment by mbtamuli** [comment]
@javian Is there anything remaining for this to get merged?

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Looking good to me! @javian letting you squash and merged this :)

**Comment by mbtamuli** [comment]
Thanks for all your suggestions to get this merged. :)
